provider/cmd/pulumi-tfgen-aws/main.go

```diff
@@ -28,5 +28,11 @@ func main() {
 		removeUnusedQuicksightTypes(spec)
 	}
 
+	defer func() {
+		for _, f := range aws.PostTfgenHook {
+			f()
+		}
+	}()
```

Comment:
This looks a little odd - why is it here?

Reply:
We need a mechanism to do things after generation has happened: We need to write out the new `replacements.json` with added TODOs.